private/buf/bufgen/generator.go

```diff
@@ -294,6 +294,21 @@ func (g *generator) execRemotePlugin(
 	if len(responses) != 1 {
 		return nil, fmt.Errorf("unexpected number of responses received, got %d, wanted %d", len(responses), 1)
 	}
+	pluginService, err := g.registryProvider.NewPluginService(ctx, remote)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create plugin service for remote %q: %w", remote, err)
+	}
+	plugin, err := pluginService.GetPlugin(ctx, owner, name)
+	if err != nil {
+		return nil, err
+	}
+	if plugin.Deprecated {
+		warnMsg := fmt.Sprintf(`Plugin "%s/%s/%s" is deprecated`, remote, owner, name)
+		if plugin.DeprecationMessage != "" {
+			warnMsg = fmt.Sprintf("%s: %s", warnMsg, plugin.DeprecationMessage)
+		}
+		g.logger.Sugar().Warn(warnMsg)
+	}
```

Comment:
This is happening after the plugin has been used for the generate service... that is probably fine since we still want the result of the generation, it's just that we want to check whether or not the plugin is deprecated? The warning message looks good though, but @amckinney can confirm :)

Reply:
Yea, we'll definitely still want to use the generation result. This looks good, but it's worth noting that it adds an additional RPC for _every_ remote plugin the user has configured in their `buf.gen.yaml`.

Fortunately all of these requests are concurrently run in the same goroutine that executes the plugin in question. It's OK to start with this - if we ever have too many RPCs, we can batch them together (just like we used to have with each plugin execution, too).